examples/bi-directional-rpc.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/progrium/qtalk-go/codec"
+	"github.com/progrium/qtalk-go/rpc"
+	"github.com/progrium/qtalk-go/rpc/rpctest"
+)
+
+type Ping struct {
+	Message string `json:"msg"`
+}
+
+func main() {
+	h := rpc.HandlerFunc(func(res rpc.Responder, call *rpc.Call) {
+		p := &Ping{}
+		if err := call.Receive(p); err != nil {
+			res.Return(fmt.Errorf("ping err: %+v", err))
+			return
+		}
+
+		res.Return(reverse(p.Message))
+	})
+
+	cli, _ := rpctest.NewPair(h, codec.JSONCodec{})
```

Comment:
this is for tests, maybe examples should avoid using those utilities. i think peer.New should get you what you want